google/cloud/storage/bucket.py

```diff
@@ -2236,6 +2237,130 @@ def rename_blob(
             )
         return new_blob
 
+    @create_trace_span(name="Storage.Bucket.moveBlob")
+    def move_blob(
+        self,
+        blob,
```

Comment:
Double checking that this is to follow the rename_blob() / copy_blob() semantics here, correct? As discussed, I guess there are both pros and cons for taking in a blob object vs blob name string

Reply:
Yes, since we have a close analogue I decided to keep the same interface.